### PR TITLE
add support for --sanity-check-only + also run sanity check for extensions when using --module-only

### PR DESCRIPTION
**Pull request by boegel**

### CODE REVIEW

File: easybuild/framework/easyblock.py [hunk 1]
```diff
@@ -3119,9 +3130,14 @@ def skip_step(self, step, skippable):
...
             skip = True
 
+        elif sanity_check_only and step != SANITYCHECK_STEP:
+            self.log.info("Skipping %s step because of sanity-check-only mode", step)
+            skip = True
+
         else:
-            self.log.debug("Not skipping %s step (skippable: %s, skip: %s, skipsteps: %s, module_only: %s, force: %s",
-                           step, skippable, self.skip, self.cfg['skipsteps'], module_only, force)
+            msg = "Not skipping %s step (skippable: %s, skip: %s, skipsteps: %s, module_only: %s, force: %s, "
```
Comment (bartoldeman) [inline easybuild/framework/easyblock.py:3138]: just wondering, for the `msg +=` is there any style that's better, e.g. 
```
msg = ("Not skipping %s step (skippable: %s, skip: %s, skipsteps: %s, module_only: %s, force: %s, "
            "sanity_check_only: %s)")
```
or  using `''.join()` or something else?

Also there is a `)` missing in the string itself, behind the last `%s`.

**Review by bartoldeman — CHANGES_REQUESTED** [review]

**Comment by boegel** [comment]
This probably needs more work to ensure that sanity checks for extensions are also done when using `eb --sanity-check-only`...

**Comment by ocaisa** [comment]
> This probably needs more work to ensure that sanity checks for extensions are also done when using `eb --sanity-check-only`...

Yes, I just noticed that in general sanity checks for extensions are not run when using `--module-only` so this is a more general shortcoming.

File: easybuild/framework/easyblock.py [hunk 2]
```diff
@@ -3119,9 +3130,14 @@ def skip_step(self, step, skippable):
             self.log.info("Skipping %s step because of forced module-only mode", step)
             skip = True
 
+        elif sanity_check_only and step != SANITYCHECK_STEP:
+            self.log.info("Skipping %s step because of sanity-check-only mode", step)
+            skip = True
```
[the remark this answers is not in the document]
Comment (boegel) [reply easybuild/framework/easyblock.py:3135]: I prefer to assume that there won't be any skipping, and only setting `skip` to `True` when we do want to skip.

Seems less likely to introduce bugs like that to me...

File: easybuild/framework/easyblock.py (hunk 1 above)
Comment (boegel) [reply easybuild/framework/easyblock.py:3138]: We use the `+=` approach all throughout the code, I think it's clearer sticking to that...

File: easybuild/framework/easyblock.py [hunk 3]
```diff
@@ -2332,6 +2309,44 @@ def extensions_step(self, fetch=False, install=True):
...
+
+        self.prepare_for_extensions()
+
+        if fetch:
+            self.exts = self.fetch_extension_sources()
+
+        self.exts_all = self.exts[:]  # retain a copy of all extensions, regardless of filtering/skipping
+
+        # actually install extensions
+        self.log.info("Installing extensions")
```
Comment (ocaisa) [inline easybuild/framework/easyblock.py:2341]: I guess this should be guarded?
```suggestion
        if install:
            self.log.info("Installing extensions")
```

File: easybuild/framework/easyblock.py [hunk 4]
```diff
@@ -3281,9 +3315,14 @@ def skip_step(self, step, skippable):
...
 
+        elif sanity_check_only and step != SANITYCHECK_STEP:
+            self.log.info("Skipping %s step because of sanity-check-only mode", step)
+            skip = True
+
         else:
-            self.log.debug("Not skipping %s step (skippable: %s, skip: %s, skipsteps: %s, module_only: %s, force: %s",
-                           step, skippable, self.skip, self.cfg['skipsteps'], module_only, force)
+            msg = "Not skipping %s step (skippable: %s, skip: %s, skipsteps: %s, module_only: %s, force: %s, "
+            msg += "sanity_check_only: %s"
```
Comment (akesandgren) [inline easybuild/framework/easyblock.py:3324]: Still missing the end ")" in that msg string

**Review by akesandgren — CHANGES_REQUESTED** [review]

**Review by akesandgren — APPROVED** [review]
LGTM

File: test/framework/toy_build.py [hunk 5]
```diff
@@ -1682,6 +1682,58 @@ def test_module_only(self):
...
+            "exts_list = [",
+            "    ('barbar', '0.0', {",
+            "        'start_dir': 'src',",
+            "        'exts_filter': ('ls -l lib/lib%(ext_name)s.a', ''),",
+            "    })",
+            "]",
+        ])
+        write_file(test_ec, test_ec_txt)
+
+        # clean up $MODULEPATH to only modules in test prefix dir are found
```
Comment (ocaisa) [inline test/framework/toy_build.py:1712]:
```suggestion
        # clean up $MODULEPATH so only modules in test prefix dir are found
```

**Review by ocaisa — CHANGES_REQUESTED** [review]
LGTM, typo in a comment

**Pull request merged**